File: server/models/Pill.js
const { DataTypes } = require('sequelize');
const sequelize = require('./database');

// Player pills
const Pill = sequelize.define('Pill', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  userId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'Users',
      key: 'id'
    }
  },
  pillId: {
    type: DataTypes.STRING,
    allowNull: false
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  effect: {
    type: DataTypes.JSON,
    allowNull: true
  }
});

module.exports = Pill;

File: server-go/internal/models/pill.go
package models

import "gorm.io/datatypes"

// Pill 对应 Sequelize Pill 模型

type Pill struct {
	ID     uint `gorm:"primaryKey;column:id"`
	UserID uint `gorm:"column:userId"`

	PillID string `gorm:"column:pillId"`
	Name   string `gorm:"column:name"`

	Description string         `gorm:"column:description"`
	Effect      datatypes.JSON `gorm:"column:effect"`
}

func (Pill) TableName() string {
	return "Pills"
}
